src/client/datascience/interactive-window/interactiveWindowProvider.ts

```diff
@@ -95,6 +95,9 @@ export class InteractiveWindowProvider implements IInteractiveWindowProvider, IA
             result = this.create(resource, mode);
         }
 
+        // make sure the existing IW has a kernel in case it failed to start previously
+        result.ensureKernelReadyPromise();
+
         await result.readyPromise;
```

Comment:
Could you put the 'ensureKernelReadyPromise' call into the getter for 'readyPromise'? Then this code wouldn't have to change.

Reply:
that makes sense. I didn't realize readyPromise was only called from the provider, that makes it easy to test